pylot/perception/base_perception_eval_operator.py

```diff
@@ -72,14 +72,21 @@ def on_watermark(self, timestamp, finished_indicator_stream):
             return
         assert len(timestamp.coordinates) == 1
         game_time = timestamp.coordinates[0]
-        (st, et) = self._prediction_start_end_times[self._start_time_frontier]
-        assert st == game_time, 'Incorrect frontier'
-        self._accuracy_compute_buffer.append((st, et, False))
-        index = self._start_time_frontier
+        on_new_prediction = self._start_time_frontier == len(
```

Comment:
Using len(self._prediction_start_end_times) might be buggy because we don't have any guarantee that we're not going to have any entries for timestamps t  + 1, t + 2.. in self._prediction_start_end_times when the watermark callback for timestamp t is invoked.

Do we need to deal with the missing start times here? or should we just assume that these are the responsibility of the upstream operator. IMO, we should let the upstream operators decide what the policy is (e.g.,output previous detections, yada yada).

Reply:
Good catch.

As for the second part, that is what the original sequencer was doing. As it stands this'd require additions to the sequencer as well as at the tracker down the line once it skips frames, creating a common function implemented in multiple places. This makes me inclined to do it in this module.

I pushed the changes to address the new comments, and they pass the tests.